### PR TITLE
Settings._let will pop element it added instead of just the last one.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/core/settings.py [hunk 1]
```diff
@@ -79,6 +79,64 @@ def fn(debug):
...
+    def pop(self):
+        if self.prv is not None:
+            self.prv.nxt = self.nxt
+
+        if self.nxt is not None:
+            self.nxt.prv = self.prv
+
+        if self.stack.end is self:
+            self.stack.end = self.prv
+            self.stack.len -= 1
```
Comment (lostella) [inline src/gluonts/core/settings.py]: Should the stack length be decreased unconditionally?

**Review by lostella — COMMENTED** [review]
Do we need our own custom `LinkedStack`? Wouldn't a regular Python `list` do the job?

**Comment by jaheba** [comment]
In short no, let me add some more docs to the PR, to argue why the the approach is needed.

**Comment by jaheba** [comment]
I wonder if we want to remove support for item-assignment at all. If you want to alter the `setting`, you should only use `push` and `let`.

**Review by lostella — APPROVED** [review]

**Pull request merged**